atom/browser/api/atom_api_desktop_capturer.cc

```diff
@@ -165,8 +165,12 @@ void DesktopCapturer::UpdateSourcesList(DesktopMediaList* list) {
       std::vector<std::string> device_names;
       // Crucially, this list of device names will be in the same order as
       // |media_list_sources|.
-      webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
-          &device_names);
+      if (!webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
+              &device_names)) {
+        Emit("error", "Failed to get sources.");
```

Comment:
I do not think we want to emit an error here, we might still want to return the TYPE_WINDOW sources, even if user is just asking TYPE_SCREEN we should just return an empty list.